pkg/collector/containerdlogs_collector.go

```diff
@@ -0,0 +1,106 @@
+package collector
+
+import (
+	"encoding/json"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/Azure/aks-periscope/pkg/interfaces"
+	"github.com/Azure/aks-periscope/pkg/utils"
+)
+
+// ContainerDLogsCollector defines a ContainerDLogs Collector struct
+type ContainerDLogsCollector struct {
+	BaseCollector
+}
+
+// ContainersList defines configuration for list of containers and their details
+type ContainersList struct {
+	ContainersList []ContainerInformation `json:"containers"`
+}
+
+// ContainerInformation defines configuration of information available for each container
+type ContainerInformation struct {
+	ID     string            `json:"id"`
+	Labels map[string]string `json:"labels"`
+}
+
+var _ interfaces.Collector = &ContainerDLogsCollector{}
+
+// NewContainerDLogsCollector is a constructor
+func NewContainerDLogsCollector(exporter interfaces.Exporter) *ContainerDLogsCollector {
+	return &ContainerDLogsCollector{
+		BaseCollector: BaseCollector{
+			collectorType: ContainerDLogs,
+			exporter:      exporter,
+		},
+	}
+}
+
+// Collect implements the interface method
+func (collector *ContainerDLogsCollector) Collect() error {
+	containerLogs := strings.Fields(os.Getenv("DIAGNOSTIC_CONTAINERLOGS_LIST"))
+	rootPath, err := utils.CreateCollectorDir(collector.GetName())
+	if err != nil {
+		return err
+	}
+
+	output, err := utils.RunCommandOnHost("crictl", "ps", "-a", "-o", "json")
+	if err != nil {
+		return err
+	}
+
+	var containers ContainersList
+	if err = json.Unmarshal([]byte(output), &containers); err != nil {
+		log.Fatalf("unable to read containers json file: %v", err)
+	}
+
+	for _, containerLog := range containerLogs {
+		containerLogParts := strings.Split(containerLog, "/")
+		targetNameSpace := containerLogParts[0]
+		containerNames := map[string]string{}
+
+		for _, container := range containers.ContainersList {
+			containerID := container.ID
+			containerNameSpace, ok := container.Labels["io.kubernetes.pod.namespace"]
+			if !ok {
+				log.Printf("Unable to retrieve namespace of container: %+v", containerID)

```

Comment:
Minor: consistency: see line 75 -->  https://github.com/Azure/aks-periscope/pull/73/files#diff-a767a7e0a1997c77cf6883c7707d9559ab33c388e714275ffd439509335825f3R75 - message starts with upper case.

Reply:
Sorry, I'm a bit confused by this comment? Should they both start with lowercase?

Reply:
`fmt.Errorf` should start with a lowercase and use %w to wrap errors
`log.Printf` should start with a capital letter and you can't use %w there. So you did good here, nothing to change :)